### ISSUE TITLE
editorconfig support

### ISSUE THREAD

**Issue by xperiandri** [dotnet/fsharp#3962]
https://github.com/Microsoft/visualfsharp/issues/2693#issuecomment-289200984
> The answer is simple as well: limited resources. While Visual F# is a Microsoft product most of the stuff in this area is done by the open source community. There is a lot happening around tooling at the moment and I assume someone will take care of formatting eventually. But right now it seems people's interests/priorities are at other (important) features. TBH there is very little hope that Microsoft will implement formatting, so our best shot is the community. You can be an active part here and help to make it happen. ;-)

#3542

@Pilchie, @sharwell is it possible to push a prototype of styling rules integration for F# so that community can pick up and add rules using some reference sample?

**As without styling and code analyzers F# will still miss critical team collaboration features.**

**Comment by cartermp** [comment]
Proper editorconfig support would require a source code formatter for F#, so this is preceded by #3542 and the work that needs to happen there being merged before we can think of a prototype for adhering to editorconfig.

**Comment by brettrowberry** [comment]
Fantomas fills this role now, right?

**Comment by cartermp** [comment]
Yes, fantomas largely does. Although this feature is meant to embed formatting into the F# compiler service itself, which would likely involve folding Fantomas into the compiler service itself. That's probably the ultimately long-term goal, but things are mostly okay where they stand for now.
